app/models/mixins/supports_feature_mixin.rb

```diff
@@ -68,19 +116,32 @@ def unsupported
   end
 
   class_methods do
-    def unsupported_reason(feature)
-      public_send("supports_#{feature}?") unless unsupported.key?(feature)
-      unsupported[feature]
-    end
-
+    # This is the DSL used a class level to define what is supported
     def supports(feature, &block)
+      SupportsFeatureMixin.guard_queryable_feature(feature)
       send(:define_supports_methods, feature, true, &block)
```

Comment:
Oh, you were totally right and I changed that in https://github.com/ManageIQ/manageiq/pull/10980